src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala

```diff
@@ -105,20 +110,140 @@ class AnxAnalyses(format: AnxAnalysisStore.Format) {
   private[this] val reader = new ProtobufReaders(mappers.getReadMapper, schema.Version.V1)
   private[this] val writer = new ProtobufWriters(mappers.getWriteMapper)
 
+  def sortProjection(value: Projection): Projection = {
+    value.copy(
+      id = value.id,
+      prefix = value.prefix,
+    )
+  }
+
+  // This should be done sometime
+  // But I can't figure out how to sort a Type differently based on which subclass it's a member of
+  def sortType[A <: Type](value: A): Type = {
+    value
+  }
+
+  def sortAnnotation(annotation: Annotation): Annotation = {
+    annotation.copy(
+      arguments = annotation.arguments.sortWith(_.hashCode() > _.hashCode())
+    )
+  }
+
+  def sortTypeParameter(typeParameter: TypeParameter): TypeParameter = {
+    typeParameter.copy(
+      annotations = typeParameter.annotations.map(annotation => sortAnnotation(annotation)).sortWith(_.hashCode() > _.hashCode()),
+      typeParameters = sortTypeParameters(typeParameter.typeParameters),
+    )
+  }
+
+  def sortStructure(structure: Structure): Structure = {
+    structure.copy(
+      parents = structure.parents.sortWith(_.hashCode() < _.hashCode()),
+      declared = structure.declared.sortWith(_.hashCode() < _.hashCode()),
+      inherited = structure.inherited.sortWith(_.hashCode() < _.hashCode()),
+    )
+  }
+
+  def sortTypeParameters(typeParameters: Seq[TypeParameter]): Seq[TypeParameter] = {
+    typeParameters.map(typeParameter => sortTypeParameter(typeParameter)).sortWith(_.hashCode() > _.hashCode())
+  }
+
+  def sortClassLike(classLike: ClassLike): ClassLike = {
+    classLike.copy(
+      annotations = classLike.annotations.map(annotation => sortAnnotation(annotation)).sortWith(_.hashCode() > _.hashCode()),
+      structure = classLike.structure.map(structure => sortStructure(structure)),
+      savedAnnotations = classLike.savedAnnotations.sorted,
+      childrenOfSealedClass = classLike.childrenOfSealedClass.map(typeInstance => sortType(typeInstance)).sortWith(_.hashCode() > _.hashCode()),
+      typeParameters = sortTypeParameters(classLike.typeParameters),
+    )
+  }
+
+  def sortCompanions(companions: Companions): Companions = {
+    Companions(
+      classApi = companions.classApi.map(classLike => sortClassLike(classLike)),
+      objectApi = companions.objectApi.map(classLike => sortClassLike(classLike)),
+    )
+  }
+
+  def sortAnalyzedClassMap(analyzedClassMap: Map[String, AnalyzedClass]): TreeMap[String, AnalyzedClass] = {
+    TreeMap[String, AnalyzedClass]() ++ analyzedClassMap.mapValues { analyzedClass =>
+      analyzedClass.copy(
+        api = analyzedClass.api.map(companions => sortCompanions(companions)),
```

Comment:
Isn't this the same as `api = analyzedClass.api.map(sortCompanions)`?
Same with the two `.map` calls in `sortCompanions` above.